nmigen_boards/upduino.py

```diff
@@ -0,0 +1,47 @@
+import os
+import subprocess
+
+from nmigen.build import *
+from nmigen.vendor.lattice_ice40 import *
+from nmigen_boards.resources import *
+
+######################################################
+# Board definition file for Gnarly Grey"s "Upduino". #
+######################################################
+
+__all__ = ["UpduinoPlatform"]
```

Comment:
Do all three of the Upduino boards have compatible pinouts? If not it would be best to call the platform `UpduinoV2Platform` or something like that.

Reply:
Good point; I think that this file should apply to V1, V2, and V2.1 boards - I double-checked the LEDs and connector pinouts on the V1 schematic, and they looked identical. I think the main difference is the lack of a USB bridge to program the flash:

https://github.com/gtjennings1/UPDuino_v1_0

And TinyVision's repository's readme has a list of differences between their V2.1 board and the V2.0; it doesn't look like there are any changes to the pin numbers or clocks:

https://github.com/tinyvision-ai-inc/UPduino-v2.1

Happy to change it, though; someone might decide to add an optional entry for the normally-disconnected oscillator on the 'v2' board later or something.

Reply:
Sounds good to me.